source/framework/vx_copy_swap.c

```diff
@@ -0,0 +1,503 @@
+                if ((vx_enum)VX_KERNEL_COPY == node->kernel->enumeration)
+                {
+                    vx_uint32 j;
+                    for (j = 0; j < num_copy_move_nodes && removable; ++j)
+                    {
+                        if (i != j)
+                        {
+                            if (first == graph->nodes[copy_move_indices[j]]->parameters[0])
+                            {
+                                removable = (vx_bool)vx_false_e;
+                            }
+                        }
+                    }
+                    tivx_obj_desc_node_t * obj_desc = (tivx_obj_desc_node_t *)(node->obj_desc[0]);
+                    for (j = 0; j < obj_desc->num_in_nodes && removable; j++)
+                    {
+                        vx_node in_node = (vx_node)ownReferenceGetHandleFromObjDescId(obj_desc->in_node_id[j]);
+                        if (((tivx_obj_desc_node_t *)(in_node->obj_desc[0]))->num_out_nodes > 1)
+                        {
+                            vx_uint32 k;
+                            for (k = 0; k < graph->num_nodes && removable; ++k)
+                            {
+                                if (copy_move_indices[i] != k)
+                                {
+                                    vx_node othernode = graph->nodes[k];
+                                    vx_uint32 l;
+                                    for (l = 0; l < othernode->kernel->signature.num_parameters && removable; ++l)
+                                    {
+                                        if ((othernode->parameters[l] == second) &&
+                                            ((vx_enum)VX_BIDIRECTIONAL == othernode->kernel->signature.directions[l]))
+                                        {
+                                            removable = vx_false_e;
+                                        }
+                                    }
+                                }
+                            }
+                        }
+                    }
+                }
+                if ((vx_bool)vx_true_e == removable)
+                {
+                    /* We test for replacing the output node first. */
+                    if ((vx_bool)vx_true_e == ownCopyMoveReplaceOutputCompatible(first, second))
+                    {
+                        /* we can remove this node, propagating the input reference */
+                        status = ownCopyMoveRemoveNode(graph, copy_move_indices[i], second, first);
+                        if ((vx_enum)VX_SUCCESS == status)
+                        {
+                            node_removed = (vx_bool)vx_true_e;
+                            /* and re-make node list list */
+                            num_copy_move_nodes = makeCopyMoveNodeList(graph, copy_move_indices);
+                            /* restart the loop */
+                            break;
+                        }
+                    }
+                    else
```

Comment:
Shall we remove the empty else clause here since this is a simple "if" condition above it and not an "if else"?  Just flagging as a code coverage gap which may be unnecessary.

Reply:
empty else removed